composer.json

```diff
@@ -36,11 +36,13 @@
         "doctrine/cache": "^1.11|^2.0",
         "doctrine/deprecations": "^0.5.3",
         "doctrine/event-manager": "^1.0",
-        "psr/cache": "^1|^2|^3"
+        "psr/cache": "^1|^2|^3",
+        "psr/log": "^1|^2"
```

Comment:
For some reason, allowing `psr/log: ^3` causes the downgrade of `symfony/console` to `v5.3.2`:
```
$ composer why-not symfony/console 5.3.10
symfony/console  v5.3.10  conflicts  psr/log (>=3)
```
In turn, this causes a [build failure](https://github.com/doctrine/dbal/runs/4205444243) on PHP 8.1 due to the issue fixed in https://github.com/symfony/console/commit/9a4fcecbcccf4b55f4c28e1c71e3cd83d7be99b8:
> Return type of Symfony\Component\Console\Helper\HelperSet::getIterator() should either be compatible with IteratorAggregate::getIterator(): Traversable, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice

This is weird because `symfony/console` requires `psr/log: ^1|^2` only as a development dependency.

Reply:
The idea to introduce a conflict of `symfony/console` with `psr/log: >= 3` seems brutal: Symfony Console requires PSR Log only as a development dependency (i.e. doesn't depend on it in production) but the conflict affects all packages that use the log in production.

Reply:
> Symfony Console requires PSR Log only as a development dependency

The package bundles a logger implementation that is incompatible with psr/log 3. That implementation is meant for production use as well.

The fact that Composer downgrades symfony/console is of course unfortunate. This will be resolved as soon as Symfony 6.0 is stable, I guess. But if we merge this PR as is, DBAL will block the installation of psr/log 3 as well. Somehow, we need to break out of this vicious circle. 🤔

Reply:
> The package bundles a logger implementation that is incompatible with psr/log 3. That implementation is meant for production use as well.

Is the logger bundled by Symfony named `\Psr\Log`? Could you provide some more details? I believe, for testing purposes, we can artificially downgrade `psr/log` to 2 when testing on PHP 8.1 to allow the installation of the newest stable Symfony Console.

Reply:
> Is the logger bundled by Symfony named `\Psr\Log`? Could you provide some more details?

The class is `Symfony\Component\Console\Logger\ConsoleLogger`. It extends `AbstractLogger` from the PSR package, but does not declare the return types that v3 requires.

> I believe, for testing purposes, we can artificially downgrade `psr/log` to 2 when testing on PHP 8.1 to allow the installation of the newest stable Symfony Console.

Let's do this then.